src/model/cli/cli.ts

```diff
@@ -3,55 +3,85 @@ import { BuildParameters, CloudRunner, ImageTag, Input } from '..';
 import * as core from '@actions/core';
 import { ActionYamlReader } from '../input-readers/action-yaml';
 import CloudRunnerLogger from '../cloud-runner/services/cloud-runner-logger';
-import { CliFunction, GetAllCliModes, GetCliFunctions } from './cli-decorator';
-import { RemoteClientLogger } from './remote-client/remote-client-services/remote-client-logger';
-import { CloudRunnerState } from '../cloud-runner/state/cloud-runner-state';
-import { SetupCloudRunnerRepository } from './remote-client/setup-cloud-runner-repository';
-import * as SDK from 'aws-sdk';
+import CloudRunnerQueryOverride from '../cloud-runner/services/cloud-runner-query-override';
+import { CliFunction, CLIFunctionsRepository } from './cli-functions-repository';
+import { AWSCLICommands } from '../cloud-runner/cloud-runner-providers/aws/commands/aws-cli-commands';
+import { Caching } from '../cloud-runner/remote-client/caching';
+import { LFSHashing } from '../cloud-runner/services/lfs-hashing';
+import { SetupCloudRunnerRepository } from '../cloud-runner/remote-client/setup-cloud-runner-repository';
 
 export class CLI {
-  static async RunCli(options: any): Promise<void> {
-    Input.githubInputEnabled = false;
-
-    const results = GetCliFunctions(options.mode);
-
-    if (results === undefined || results.length === 0) {
-      throw new Error('no CLI mode found');
-    }
-
-    CloudRunnerLogger.log(`Entrypoint: ${results.key}`);
-
-    options.versioning = 'None';
-    Input.cliOptions = options;
-    return await results.target[results.propertyKey]();
+  public static options;
+  static get cliMode() {
+    return CLI.options !== undefined && CLI.options.mode !== undefined && CLI.options.mode !== '';
   }
-  static isCliMode(options: any) {
-    return options.mode !== undefined && options.mode !== '';
+  public static query(key, alternativeKey) {
+    if (CLI.options && CLI.options[key] !== undefined) {
+      return CLI.options[key];
+    }
+    if (CLI.options && alternativeKey && CLI.options[alternativeKey] !== undefined) {
+      return CLI.options[alternativeKey];
+    }
+    return;
   }
```

Comment:
`Cli.query` is very abstract in naming. 

Would it make sense to use a common `options` approach where you have overrides that are passed in and then there are no "alternative keys"?

Furthermore, I'd argue that even something like `getOption` would be quite abstract still, and far away from the domain. Abstract code is both difficult to understand and maintain.

An example that could be a bit closer to the domain could look like the following:

Let's assume `key` that's passed in would be `runner`

```ts
private get isUsingGitHubRunner() {
  return this.options.runner === 'github'
}
```

Subsequent code becomes much easier to read:

```ts
if (this.isUsingGitHubRunner) {
  // something specific to this option
}
```

Reply:
Agreed. 

When I started this work it was not clear the extent of the complexity of the configuration domain would be needed in total.

This is a fully dynamic / generic implementation.

We can harden to a specific implementation somewhat. But it does relate to a super set of input. 

So it would  potentially benefit from alterations to input as well, as in encapsulating it within a wider configuration concept, and this also made me want to hold off to start with.

Reply:
Whatever you think is best here. We can always iterate as well.

Reply:
Created an issue to follow up on this and address it thoroughly